background.tex

```diff
@@ -3,62 +3,63 @@
-Even though the proposed solutions to avoid stragglers are specifically applied to the analysis of MD trajectories in the \package{MDAnalysis} library, all of these principles and implementation techniques are potentially applicable to data analysis in other Python-based libraries.
+In the present study, we identify the cause of the straggler tasks in the context of analyzing large MD trajectories and provide solutions through which we can improve performance and scaling.
+Even though the proposed solutions to avoid stragglers are specifically applied to the analysis of MD trajectories in \package{MDAnalysis}, all of these principles and implementation techniques are potentially applicable to data analysis in other Python-based libraries.
 
 \subsection{Other packages with parallel analysis capabilities}
 \label{sec:otherparallel}
 
 There have been various attempts to parallelize the analysis of MD trajectories. 
-HiMach \cite{himach-2008} introduces scalable and flexible parallel Python framework to deal with massive MD trajectories, by combining and extending the strengths of Google's MapReduce and the VMD analysis tool \cite{Hum96}. 
-HiMach's runtime is responsible to parallelize and distribute the Map and Reduce classes to the assigned cores.
-HiMach uses parallel I/O for file access during the map tasks and \text{MPI\_Allgather} in the reduction process. 
-HiMach, however, does not discuss parallel analysis for the task types which can not be analyzed using a MapReduce implementation.
-The analysis needs by MD community, however, tend to be varied and a generalized parallel framework should be able to provide parallel analysis for a variety of task types.
-Furthermore, HiMach is not available under an open source license, which makes it difficult to improve upon this work or integrate it in more complex analysis workflows.
-
-Wu et. al. \cite{Wu_et.al} present a scalable parallel framework for distributed-memory post-simulation data analysis.
+HiMach~\cite{himach-2008} introduces scalable and flexible parallel Python framework to deal with massive MD trajectories, by combining and extending Google's MapReduce and the VMD analysis tool~\cite{Hum96}. 
+HiMach's runtime is responsible to parallelize and distribute Map and Reduce classes to assigned cores.
+HiMach uses parallel I/O for file access during map tasks and \text{MPI\_Allgather} in the reduction process. 
+HiMach, however, does not discuss parallel analysis of analysis types that cannot be implemented via MapReduce.
+\gpnote{I do not understand this sentence at all. I do not think it is needed at all here. I suggest to remove it. :
+The analysis needs by MD community, however, tend to be varied and a generalized parallel framework should be able to provide parallel analysis for a variety of task types.}
```

Comment:
I am not sure about the purpose of this sentence. Does it want to argue about the fact that the analysis methods are numerous and a generalized library is needed?

Isn't that a bit out of the scope of the paper since we are discussing parallel implementations of analysis algorithms and not whether there is a generalized framework or not?

Reply:
Yes, basically I was trying to say HiMach is suitable for a specific type of analysis method (Only MapReduce). At the same time, we want to convince the reader that the factors that are separating our work from the rest are the general guidelines that we provide and are not necessarily limited too MapReduce analysis approach.

Reply:
Then why not say it the way you just said it?

Reply:
Then please go ahead and modify the text in the way you think is good and then we will merge it into the master branch.